### PR TITLE
Basic AWS Lambda setup

### PR DESCRIPTION
**Pull request by spydersweb**
Trying to get back to where i was

### CODE REVIEW

File: terraform/lambda.tf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module "lambda" {
+  source = "./modules/lambda"
+
+  function_name = "hello-world"
```
Comment (spydersweb) [inline terraform/lambda.tf]: This is where I am getting the problem, when I plan this module I get "function_name" not expected here

File: terraform/modules/lambda/locals.tf [hunk 2]
```diff
@@ -0,0 +1,8 @@
+locals {
+  src_path     = "${path.module}/lambda/${var.function_name}"
```
Comment (spydersweb) [inline terraform/modules/lambda/locals.tf]: If I terraform init this module I get the same error message, which is "reference to undeclared input variable", and yet they are declared in variables.tf.  What am I missing!!!

File: terraform/modules/lambda/variables.tf [hunk 3]
```diff
@@ -0,0 +1,12 @@
+variable "function_name" {
```
Comment (spydersweb) [inline terraform/modules/lambda/variables.tf:1]: These are the input variable declarations and when I try and create the module the IDE does recognise their existence

File: terraform/modules/lambda/main.tf [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+data "archive_file" "function_archive" {
+  depends_on = [null_resource.function_binary]
+
+  type        = "zip"
+  source_file = local.binary_path
+  output_path = local.archive_path
+  }
+
+resource "aws_lambda_function" "function" {
+  function_name = var.function_name
```
Comment (spydersweb) [inline terraform/modules/lambda/main.tf:16]: Where ever I reference a var.<name> I get a declaration error

File: terraform/modules/lambda/iam.tf [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      resources = [
+        "arn:aws:logs:*:*:*",
+      ]
+    }
+}
+
+resource "aws_iam_policy" "function_logging_policy" {
+  name = "AllowLambdaLoggingPolicy"
+  description = "Policy for lambda cloudwatch logging"
+  policy = data.aws_iam_policy_document.allow_lambda_logging
```
Comment (WJDAVIES) [inline terraform/modules/lambda/iam.tf]: this will need to be -->  policy = data.aws_iam_policy_document.allow_lambda_logging.json

**Pull request merged**